frontend/src/app/helpers/editor/editor.component.ts

```diff
@@ -3,71 +3,93 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 import {
-  AfterViewInit,
   Component,
+  computed,
+  effect,
   ElementRef,
   EventEmitter,
   HostListener,
-  Input,
-  NgZone,
+  input,
+  model,
   Output,
-  ViewChild,
+  signal,
 } from '@angular/core';
-import * as monaco from 'monaco-editor';
+import { FormsModule } from '@angular/forms';
+import { type editor } from 'monaco-editor';
+import { MonacoEditorModule, NgxEditorModel } from 'ngx-monaco-editor-v2';
 import { ToastService } from 'src/app/helpers/toast/toast.service';
 import { v4 as uuidv4 } from 'uuid';
+import { URI } from 'vscode-uri';
 import { stringify, parse, YAMLParseError } from 'yaml';
 
 @Component({
   selector: 'app-editor',
   templateUrl: './editor.component.html',
   standalone: true,
+  imports: [MonacoEditorModule, FormsModule],
 })
-export class EditorComponent implements AfterViewInit {
-  @Input()
-  height = '400px';
-
-  @Input()
-  lineWidth = 55;
-
-  @Input()
+export class EditorComponent {
+  height = input('400px');
+  lineWidth = input(55);
   // Helps to identify the editor in the DOM
-  context = uuidv4();
-
-  private editor?: monaco.editor.IStandaloneCodeEditor = undefined;
-  intialValue = 'Loading...';
-
-  @ViewChild('editorRef') editorRef: ElementRef | undefined;
+  context = input(uuidv4());
+  value = input<unknown>();
+  editor = signal<editor.IStandaloneCodeEditor | null>(null);
+
+  type = input<'globalconfig' | 'tool' | 'toolnature' | 'toolversion'>();
+
+  initialValue = signal('Loading...');
+  code = model(this.initialValue());
+
+  options: editor.IStandaloneEditorConstructionOptions = {
+    language: 'yaml',
+    minimap: { enabled: false },
+    overviewRulerBorder: false,
+    scrollBeyondLastLine: false,
+    automaticLayout: true,
+    tabSize: 2,
+    fixedOverflowWidgets: true,
+  };
+
+  editorModel = computed<NgxEditorModel>(() => ({
+    value: this.initialValue(),
+    language: 'yaml',
+    uri: URI.parse(`file:///${this.context()}.${this.type()}.yaml`),
```

Comment:
```suggestion
    uri: `file:///${this.context()}.${this.type()}.yaml`,
```

And get rid of `vscode-uri`, which seems to imported just for this case.

Reply:
Without `vscode-uri` error highlighting doesn't work, and there are error logs in the browser console